### ISSUE TITLE
"accept from source" with authentication?

### ISSUE THREAD

**Issue by jootuom** [OpenSMTPD/OpenSMTPD#757]
It seems you're unable to use "accept from source" if you also have authentication enabled on the listener.
Therefore you're only able to use authentication OR IP-based access.

Intended or a bug?

I tested with the following config:
```
pki host.domain.tld certificate "/etc/opensmtpd/pki/host.domain.tld.crt"
pki host.domain.tld key "/etc/opensmtpd/pki/host.domain.tld.key"
table sources file:/etc/opensmtpd/sources
table users file:/etc/opensmtpd/users

listen on eth0 port 25 tls auth <users>
listen on eth0 port 587 tls auth <users>

accept from source <sources> for any relay via tls://relay.domain.tld
```

Of course, I have the IP address of the SMTP client in the sources table in "192.0.2.1" format, not a subnet.

It will always result in "550 Invalid recipient" after "RCPT TO:".
Changing the accept line to `accept from local` makes this work fine - without the IP-based access control of course.

Tested on CentOS 7 with the OpenSMTPD EPEL package, version 6.0.2p1

**Comment by poolpOrg** [comment]
It is an unintended side-effect of an intended feature.

Basically, once a client authenticated, it is considered just like if the user had logged in locally and send the mail from the local host, therefore it matches 'from local' and no longer the source address.

We have plans to rework how sessions are flagged, I'll have a look this week to see if there's an easy fix that can be applied before the refactor

**Comment by poolpOrg** [comment]
Discussed with @ericfaurot, this will be very tricky to fix with current configuration, we will slip a fix for this while working on https://github.com/OpenSMTPD/OpenSMTPD/issues/772

**Comment by poolpOrg** [comment]
This issue has been solved while working on #772, I'll close your ticket because it is now referenced in the other ticket and I'll track progress of the config rewrite there.

The fix will not be included in OpenSMTPD 6.2 because it's too close, however it will be part of snapshots we publish right after 6.2 release and will be included as part of OpenSMTPD 6.3
